framework/db/pgsql/Schema.php

```diff
@@ -108,7 +108,7 @@ class Schema extends \yii\db\Schema
         'unknown' => self::TYPE_STRING,
 
         'uuid' => self::TYPE_STRING,
-        'json' => self::TYPE_STRING,
+        'json' => self::TYPE_JSON,
         'jsonb' => self::TYPE_STRING,
```

Comment:
same should apply for jsonb, the difference is in the storage of the data, not in the data format. https://www.postgresql.org/docs/9.4/static/datatype-json.html